pkg/netlight/network.go

```diff
@@ -391,6 +389,38 @@ func (n *networker) Interfaces(iface string, netns string) (pkg.Interfaces, erro
 	return pkg.Interfaces{Interfaces: interfaces}, nil
 }
 
+func (n *networker) UnSetPublicConfig() error {
+	return public.DeletePublicConfig()
+}
+
+// Set node public namespace config
+func (n *networker) SetPublicConfig(cfg pkg.PublicConfig) error {
+	if cfg.Equal(pkg.PublicConfig{}) {
+		return fmt.Errorf("public config cannot be unset, only modified")
+	}
+
+	current, err := public.LoadPublicConfig()
+	if err != nil && err != public.ErrNoPublicConfig {
+		return errors.Wrapf(err, "failed to load current public configuration")
+	}
+
+	if current != nil && current.Equal(cfg) {
+		// nothing to do
+		return nil
+	}
+
+	if err := public.SavePublicConfig(cfg); err != nil {
+		return errors.Wrap(err, "failed to store public config")
+	}
+
+	return nil
+}
+
+func (n *networker) LoadPublicConfig() (pkg.PublicConfig, error) {
```

Comment:
no need for this method u already calling public.loadPublicConfig without changes

Reply:
this is important to implement the networker interface https://github.com/threefoldtech/zoslight/blob/main/pkg/network_light.go#L34 which is used to generate the stubs https://github.com/threefoldtech/zoslight/blob/ff8a7a63a0c9c717a3e2ceed73112e8fbf99da83/pkg/stubs/network_light_stub.go#L163 this is used by the gateway module